### ISSUE TITLE
[Feature]. Missing Constructor

### ISSUE THREAD

**Issue by PaulJPhilp** [callstack/ts-regex-builder#74]
**Is your feature request related to a problem? Please describe.**
When building complex RegExes, I often need to do something like:

```
const port = [portSeperator, portNumber];
type = const port: (string | Repeat)[]
```

**OR**

```
const hostname = [host, optional(repeat([period, host], { min: 1, max: 255 }))];
type = const hostname: (Repeat | Optional)[]
```

Then, I want to use those components in a builder:

```
const urlAuthority = [optional([userInfo, at]), hostname, optional(port)];
export const UrlAuthorityFinder = buildRegExp(urlAuthority, {
  ignoreCase: true,
  global: true,
});
```
This causes the following type error:
`Type '(Repeat | Optional)[]' is not assignable to type 'RegexElement'.ts(2345)`

To get around this I use a constructor [`capture(hostname) or choiceOf(hostname)`].  This works but is ugly.

**Describe the solution you'd like**
I'd like to see a constructor (combine?, **compose**?, noop?, **group**?, ...) that groups the elements into a RegexSequence.

```
const port = compose([portSeperator, portNumber]);

OR

const hostname = group([host, optional(repeat([period, host], { min: 1, max: 255 }))]);
```

**Describe alternatives you've considered**
It turns out that this is the use case I had for non-capture groups.  Even this is ugly.

**Checklist**
- [ ] Implementation
- [ ] Tests
- [ ] API docs 
- [ ] README docs (if relevant) 
- [ ] Example docs & tests (if relevant)

**Additional context**

**Comment by mdjastrzebski** [comment]
Swift Regex Builder uses `Regex(...)` for that, so adapting it analogously to TS Regex Builder that would be:
```
const port = regex([portSeperator, portNumber]);
```

Alternatively, we've might allow for nesting `RegexSequences` as elements of `RegexSequences` (as arrays), so that for encoding, they would be flattened.

```
const urlAuthority: RegexSequence = [optional([userInfo, at]), choiceOf(hostname), optional(port)];
export const UrlAuthorityValidator = buildRegExp(
  [startOfString, urlAuthority, endOfString], // <-- Nested array happens here
  {
    ignoreCase: true,
  },
);
```

**Comment by PaulJPhilp** [comment]
I see the benefit of both solutions, but I only need the regex() api for
the URL pattern.

Thank you,
Paul

**Comment by mdjastrzebski** [comment]
Resolved by #77

**Comment by mdjastrzebski** [comment]
Resolved in [v1.4.0](https://github.com/callstack/ts-regex-builder/releases/tag/v1.4.0) 🎉
